test/distributed/org/apache/cassandra/distributed/upgrade/CompactionHistorySystemTableUpgradeTest.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.distributed.upgrade;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import com.vdurmont.semver4j.Semver;
+import org.apache.cassandra.distributed.api.ConsistencyLevel;
+import org.apache.cassandra.tools.ToolRunner;
+import org.apache.cassandra.utils.FBUtilities;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import static org.apache.cassandra.tools.ToolRunner.invokeNodetoolJvmDtest;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(Parameterized.class)
+public class CompactionHistorySystemTableUpgradeTest extends UpgradeTestBase
+{
+    @Parameterized.Parameter
+    public Semver version;
+
+    @Parameterized.Parameters()
+    public static ArrayList<Semver> versions()
+    {
+      return Lists.newArrayList(v30, v3X, v40, v41);
+    }
+
+    @Test
+    public void compactionHistorySystemTableTest() throws Throwable
+    {
+        new TestCase()
+            .nodes(2)
```

Comment:
why do you need 2 nodes? compaction is a local process though

Reply:
@Maxwell-Guo this is good question, why two nodes?

Reply:
Yes one is ok for local compaction.
For the test is under a upgrade of  distributed dir （[org/apache/cassandra/distributed/upgrade/).And I think more than one node feels closer to the production system. In fact, I have three nodes at first. I think it can also cover the serial restart of nodes during the upgrade process,like gossip communctaion and so on.

Reply:
1 node is enough as it is a table in the `system` keyspace which is local and it is not distributed anywhere.